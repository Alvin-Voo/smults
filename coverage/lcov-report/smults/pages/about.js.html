<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for smults/pages/about.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">smults/pages</a> about.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';// work with components
import Layout from '../components/Layout';
&nbsp;
export default () =&gt; (
  &lt;Layout item="about"&gt;
    &lt;div className="aboutDiv"&gt;
      &lt;h2&gt;Intro video&lt;/h2&gt;
      &lt;iframe
        title="youtube-intro"
        width="750"
        height="425"
        src="https://www.youtube.com/embed/SWYt4bGmN7w"
        frameBorder="0"
        allow="autoplay; encrypted-media"
        allowFullScreen
      /&gt;
    &lt;/div&gt;
    &lt;div className="aboutDiv"&gt;
      &lt;h3&gt;Why smults?&lt;/h3&gt;
      &lt;p&gt;
        &lt;a href="https://steemit.com" target="_blank" rel="noopener noreferrer"&gt;steemit.com&lt;/a&gt;
        {' '}
        allows searches by using only ONE tag, while
        {' '}
        &lt;a href="https://busy.org/" target="_blank" rel="noopener noreferrer"&gt;busy.org&lt;/a&gt;
        {' '}
        searches by single user&amp;apos;s name,
        this is inconvenient for those who need to look for more specific posts.
        Smults is here to address this issue!
      &lt;/p&gt;
      &lt;h3&gt;Can I add custom tags in the search bar?&lt;/h3&gt;
      &lt;p&gt;Sure! Just start typing and press &amp;apos;Enter&amp;apos; to include the tag(s) when you are done!&lt;/p&gt;
      &lt;h3&gt;Why &amp;apos;15&amp;apos; trending tags to start with? Why not &amp;apos;18&amp;apos; or &amp;apos;23&amp;apos;?&lt;/h3&gt;
      &lt;p&gt;Well.. That&amp;apos;s just my personal preference ;). You can add more custom search tags by youself!&lt;/p&gt;
      &lt;h3&gt;What is this &amp;apos;Mark first tag as category&amp;apos; checkbox thing?&lt;/h3&gt;
      &lt;p&gt;When you mark it and search, the return results will have their categories as the first tag you are searching for. Isn&amp;apos;t that just convenient? ;)&lt;/p&gt;
      &lt;h3&gt;I spotted a nasty error/bug.. Where can I report it?&lt;/h3&gt;
      &lt;p&gt;
        Head over to this project&amp;apos;s
        &lt;a href="https://github.com/Alvin-Voo/smults" target="_blank" rel="noopener noreferrer"&gt;github page&lt;/a&gt;
        {' '}
        and log an issue.
      &lt;/p&gt;
      &lt;h3&gt;I have a great idea for your application! I want to contribute!&lt;/h3&gt;
      &lt;p&gt;
        Great! Let me know at
        &lt;a href="https://steemit.com/@alvinvoo" target="_blank" rel="noopener noreferrer"&gt;@alvinvoo&lt;/a&gt;
        , comment on my post which announces this project, or you can visit this project&amp;apos;s
        &lt;a href="https://github.com/Alvin-Voo/smults" target="_blank" rel="noopener noreferrer"&gt;github page&lt;/a&gt;
        {' '}
        and log an issue but with the title starting with &amp;apos;[FEATURE]&amp;apos;
      &lt;/p&gt;
      &lt;h3&gt;Is this free?&lt;/h3&gt;
      &lt;p&gt;Of course! I will let you know when it&amp;apos;s not. ;)&lt;/p&gt;
    &lt;/div&gt;
  &lt;/Layout&gt;
);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Sep 16 2018 11:24:24 GMT+0800 (+08)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
