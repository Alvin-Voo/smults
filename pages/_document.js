// ./pages/_document.js
import Document, { Head, Main, NextScript } from 'next/document';

export default class MyDocument extends Document {
  render() {
    return (
      <html>
        <Head>
          <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.1/semantic.min.css" />
          <link rel="stylesheet" href="/_next/static/style.css" />
          <link rel="shortcut icon" href="/static/favicon.ico" type="image/x-icon"></link>
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </html>
    );
  }
}
