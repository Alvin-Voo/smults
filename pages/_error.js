import React from 'react';
import PropTypes from 'prop-types';
import Error from 'next/error';

export default class MyError extends React.Component {
  static async getInitialProps({ req, res }) {
    const { statusCode } = res;

    return { statusCode };
  }

  render() {
    const { statusCode } = this.props;
    return (
      <div className="errorPage">
        <a href="/">Return to Home Page</a>
        <Error statusCode={statusCode} />
      </div>
    );
  }
}

MyError.propTypes = {
  statusCode: PropTypes.number.isRequired,
}
