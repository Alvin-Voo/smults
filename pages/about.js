import React, {Component} from 'react';//work with components
import Layout from '../components/Layout';

class About extends Component{

  render(){
    return(
      <Layout item='about'>
        About
      </Layout>
    )
  }

}

export default About;
