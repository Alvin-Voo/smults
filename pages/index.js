import React, { Component } from 'react';// work with components
import { connect } from 'react-redux';
import ConnectedSearch from './search';


export class Index extends Component {
  static getInitialProps({ reduxStore, req }) {
    // this is the initial props function in server
    const isServer = !!req;
    return { isServer };
  }

  render() {
    return <ConnectedSearch />;
  }
}

export default connect()(Index);
