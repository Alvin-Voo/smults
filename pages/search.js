import React, { Component } from 'react';// work with components
import PropTypes from 'prop-types';
import {
  Dropdown, Button, Checkbox, Popup, Icon,
} from 'semantic-ui-react';
import { take } from 'lodash';
import { connect } from 'react-redux';
import Layout from '../components/Layout';
import ConnectedPostList from '../components/PostList';
import ScrollButton from '../components/ScrollButton';
import { fetchTrendingTags, fetchPosts } from '../actions';

export class Search extends Component {
  state = {
    tagsOptions: [],
    selectedTags: [],
    filterOptions: [
      { text: 'New', value: 'created' },
      { text: 'Trending', value: 'trending' },
      { text: 'Hot', value: 'hot' },
      { text: 'Active', value: 'active' },
      { text: 'Promoted', value: 'promoted' },
    ],
    selectedFilter: 'created',
    checkedCategory: false,
    searchLoading: false,
    error: '',
  }

  componentDidMount = async () => {
    const { fetchTrendingTags } = this.props;
    await fetchTrendingTags();
    // we only want to fetch in client side once
    const { tagsOptions } = this.props;
    this.setState({ tagsOptions });
  }

  dropDownChange = (e, d) => {
    this.setState({ selectedTags: take(d.value, 5) });
  }

  dropDownAddItem = (e, d) => {
    this.setState((prevState) => {
      const newTags = prevState.tagsOptions;
      newTags.unshift({ key: d.value, value: d.value, text: d.value });
      return { tagsOptions: newTags };
    });
  }

  search = async () => {
    const { selectedTags, selectedFilter, checkedCategory } = this.state;
    const { fetchPosts } = this.props;
    if (selectedTags.length === 0) return;
    this.setState({ searchLoading: true, error: '' });
    await fetchPosts(selectedTags, selectedFilter, checkedCategory).catch(
      (error) => {
        this.setState({ error: String(error) });
      },
    );
    this.setState({ searchLoading: false });
  }

  filterChange = (e, d) => {
    this.setState({ selectedFilter: d.value });
  }

  checkBoxCategory = (e, d) => {
    this.setState({ checkedCategory: d.checked });
  }

  displayError = () => {
    const { error } = this.state;
    if (error) {
      return (
        <div className="error">
          <p>{error}</p>
          <p>
            Sorry, something is broken. Take a screenshot and notify
            <a href="https://steemit.com/@alvinvoo" target="_blank" rel="noopener noreferrer">
              {' '}
              @alvinvoo
            </a>
          </p>
        </div>
      );
    }
  }

  render() {
    const {
      tagsOptions, selectedTags, filterOptions, selectedFilter, searchLoading, checkedCategory,
    } = this.state;
    return (
      <Layout item="search">
        <div className="searchBar">
          <Dropdown
            icon="search"
            placeholder="Insert your tags here..."
            selection
            multiple
            search
            fluid
            allowAdditions
            options={tagsOptions}
            onAddItem={this.dropDownAddItem}
            onChange={this.dropDownChange}
            value={selectedTags}
          />
          <Button
            positive
            loading={searchLoading}
            onClick={this.search}
          >
            {' '}
            Search
          </Button>
        </div>
        <div className="searchOptionsBar">
          <div>
          Filter tags by
            {' '}
            {' '}
            <Dropdown
              inline
              options={filterOptions}
              onChange={this.filterChange}
              value={selectedFilter}
            />
          </div>
          <div className="markyTag">
            <Checkbox label="Mark first tag as category" onChange={this.checkBoxCategory} checked={checkedCategory} />
            <Popup trigger={<Icon name="question circle outline" />} content="Return search results which match the given first tag as their categories." />
          </div>
        </div>
        <div className="postList">
          <ConnectedPostList />
          {this.displayError()}
        </div>
        <ScrollButton scrollStepInPx={50} delayInMs={16.66} />
      </Layout>
    );
  }
}

Search.propTypes = {
  fetchPosts: PropTypes.func.isRequired,
  fetchTrendingTags: PropTypes.func.isRequired,
  tagsOptions: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
};

export function mapStateToProps({ tags }) {
  return { tagsOptions: tags.trending_tags_options };
}

export default connect(mapStateToProps, { fetchTrendingTags, fetchPosts })(Search);
