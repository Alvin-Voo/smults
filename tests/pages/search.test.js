import React from 'react';
import { shallow } from 'enzyme';
import { Search, mapStateToProps } from '../../pages/search';
import tags from '../fixtures/trendingTagsList';

/**
 * There's no direct test on the connected search component because the main
 * pain point would be dealing with the async actions. I tried jest.mock but
 * there's a complain about Unhandled promise rejection. So, I've decided to
 * just test the unconnected component and the mapStateToProps function itself.
 * Since the actions and store middlewares is already tested in the actions'
 * unit test cases
*/
describe('Testing on unconnected Search', () => {
  let wrapper;
  let fetchPosts;
  let fetchTrendingTags;
  beforeEach(async () => {
    fetchPosts = jest.fn(() => Promise.resolve(true));
    fetchTrendingTags = jest.fn(() => Promise.resolve(true));
    // shallow will call componentDidMount() which will fetchTrendingTags - which is async function
    wrapper = await shallow(
      <Search
        fetchPosts={fetchPosts}
        fetchTrendingTags={fetchTrendingTags}
        tagsOptions={[...tags]}
        // to pass only a COPY of tags, if passed direct, tagsOptions will refer to tags
      />,
    );
  });

  test('should render Search page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should fetch trending tags on mount', () => {
    expect(fetchTrendingTags).toHaveBeenCalledTimes(1);
    expect(wrapper.state('tagsOptions')).toEqual(tags);
    expect(wrapper.find('Dropdown').at(0).prop('options')).toEqual(tags);
  });

  test('should have default tags list in dropdown', () => {
    const defaultTagsOptions = wrapper.find('Dropdown').at(0).prop('options');
    expect(defaultTagsOptions).toEqual(tags);
  });

  describe('Testing tags dropdown', () => {
    test('should add tag to dropdown', () => {
      const tagToAdd = 'whale';
      wrapper.find('Dropdown').at(0).prop('onAddItem')({}, { value: tagToAdd });
      expect(wrapper.state('tagsOptions')).toEqual([{ key: tagToAdd, value: tagToAdd, text: tagToAdd }, ...tags]);
      expect(wrapper.find('Dropdown').at(0).prop('options')).toEqual([{ key: tagToAdd, value: tagToAdd, text: tagToAdd }, ...tags]);
    });

    test('should change selected tags accordingly', () => {
      const tagsToSelect = ['laundry', 'sewing machine'];
      wrapper.find('Dropdown').at(0).prop('onChange')({}, { value: tagsToSelect });
      expect(wrapper.state('selectedTags')).toEqual(tagsToSelect);
      expect(wrapper.find('Dropdown').at(0).prop('value')).toEqual(tagsToSelect);
    });

    test('should select only maximum of 5 tags', () => {
      const tagsToSelect = ['tag1', 'tag2', 'tag3', 'tag4', 'tag5', 'tag6', 'tag7'];
      wrapper.find('Dropdown').at(0).prop('onChange')({}, { value: tagsToSelect });
      expect(wrapper.state('selectedTags')).toEqual(tagsToSelect.slice(0, 5));
      expect(wrapper.find('Dropdown').at(0).prop('value')).toEqual(tagsToSelect.slice(0, 5));
    });
  });

  describe('Testing search function', () => {
    test('should search accordingly', () => {
      const tagsToSelect = [tags[2].value, tags[4].value];// kr, art
      wrapper.find('Dropdown').at(0).prop('onChange')({}, { value: tagsToSelect });
      wrapper.find('Button').simulate('click');
      expect(fetchPosts).toHaveBeenCalledTimes(1);
      expect(fetchPosts).toHaveBeenCalledWith(tagsToSelect, 'created', false);
    });

    // might have to remove very soon
    test('should not search if no tags are selected', () => {
      wrapper.find('Button').simulate('click');
      expect(wrapper.state('selectedTags')).toEqual([]);
      expect(fetchPosts).toHaveBeenCalledTimes(0);
    });

    test('should set error state and display error if fetchPost returned with error', async () => {
      const error = 'some kinda error';
      const tagsToSelect = [tags[2].value, tags[4].value];// kr, art
      // overwrite fetchPosts implementation here
      wrapper.instance().props.fetchPosts.mockImplementation(
        () => Promise.reject(new Error(error)),
      );
      wrapper.find('Dropdown').at(0).prop('onChange')({}, { value: tagsToSelect });
      // wait for ALL async functions (including async inner function) to return
      await Promise.all(wrapper.find('Button').simulate('click'));
      expect(fetchPosts).toHaveBeenCalledTimes(1);
      expect(wrapper.state('error')).toBe(`Error: ${error}`);
      expect(wrapper.find('.error')).toBeTruthy();
      expect(wrapper.find('.error > p:first-child').text()).toBe(`Error: ${error}`);
      expect(wrapper).toMatchSnapshot();
    });
  });

  /**
   * Since the search loading animation will happen from false to true to false
   * within one click, there's no way to test UI changes in stages; Splitting
   * the test into a test for state change and a test for linkage of state to UI
   * prop
   */
  describe('Testing search Button loading animation', () => {
    test('should have searchLoading state enabled while searching, and disabled after that', async () => {
      const tagsToSelect = [tags[2].value, tags[4].value];// kr, art
      wrapper.find('Dropdown').at(0).prop('onChange')({}, { value: tagsToSelect });
      wrapper.instance().setState = jest.fn();
      await Promise.all(wrapper.find('Button').simulate('click'));

      expect(wrapper.instance().setState).toBeCalledTimes(2);
      expect(wrapper.instance().setState).toHaveBeenNthCalledWith(1, { searchLoading: true, error: '' });
      expect(wrapper.instance().setState).toHaveBeenLastCalledWith({ searchLoading: false });
    });

    test('Button should have loading prop tied to searchLoading state', () => {
      wrapper.setState({ searchLoading: true });
      expect(wrapper.find('Button').prop('loading')).toBe(true);
      wrapper.setState({ searchLoading: false });
      expect(wrapper.find('Button').prop('loading')).toBe(false);
    });
  });

  describe('Testing filter dropdown', () => {
    test('should have all filter options, and default to first filter options value', () => {
      const filters = wrapper.state('filterOptions');
      expect(wrapper.find('Dropdown').at(1).prop('options')).toEqual(filters);
      const value = wrapper.state('selectedFilter');
      expect(wrapper.find('Dropdown').at(1).prop('value')).toBe(value);
    });

    test('should change filter accordingly', () => {
      const filters = wrapper.state('filterOptions');
      const { value } = filters[3];
      wrapper.find('Dropdown').at(1).prop('onChange')({}, {
        value,
      });
      expect(wrapper.state('selectedFilter')).toBe(value);
      expect(wrapper.find('Dropdown').at(1).prop('value')).toBe(value);
    });
  });

  describe('Testing Mark first tag as category', () => {
    test('should default to checkedCategory default state, which is false', () => {
      expect(wrapper.state('checkedCategory')).toBe(false);
      expect(wrapper.find('Checkbox').prop('checked')).toBe(false);
    });

    test('should change accordingly', () => {
      wrapper.find('Checkbox').prop('onChange')({}, {
        checked: true,
      });
      expect(wrapper.state('checkedCategory')).toBe(true);
      expect(wrapper.find('Checkbox').prop('checked')).toBe(true);
    });
  });
});

test('mapStateToProps should return props correctly', () => {
  const props = mapStateToProps({
    tags: {
      trending_tags_options: tags,
    },
  });
  expect(props).toEqual({
    tagsOptions: tags,
  });
});
