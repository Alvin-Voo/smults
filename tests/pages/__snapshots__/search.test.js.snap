// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on unconnected Search Testing dropdown input for author's name search should render the .authorName div with icon and dropdown when filter = author 1`] = `
<Layout
  item="search"
>
  <div
    className="searchBar"
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      allowAdditions={true}
      closeOnBlur={true}
      deburr={false}
      fluid={true}
      icon="search"
      minCharacters={1}
      multiple={true}
      noResultsMessage="No results found."
      onAddItem={[Function]}
      onChange={[Function]}
      openOnFocus={true}
      options={
        Array [
          Object {
            "text": "life",
            "value": "life",
          },
          Object {
            "text": "photography",
            "value": "photography",
          },
          Object {
            "text": "kr",
            "value": "kr",
          },
          Object {
            "text": "steemit",
            "value": "steemit",
          },
          Object {
            "text": "art",
            "value": "art",
          },
          Object {
            "text": "yatch",
            "value": "yatch",
          },
          Object {
            "text": "crypto",
            "value": "crypto",
          },
          Object {
            "text": "cake",
            "value": "cake",
          },
          Object {
            "text": "chalk",
            "value": "chalk",
          },
          Object {
            "text": "zebra",
            "value": "zebra",
          },
        ]
      }
      placeholder="Insert your tags here. Or leave empty to search for all posts..."
      renderLabel={[Function]}
      search={true}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      selection={true}
      value={Array []}
      wrapSelection={true}
    />
    <Button
      as="button"
      loading={false}
      onClick={[Function]}
      positive={true}
      role="button"
    >
       
      Search
    </Button>
  </div>
  <div
    className="searchOptionsBar"
  >
    <div>
      Filter tags by
       
       
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        deburr={false}
        icon="dropdown"
        inline={true}
        minCharacters={1}
        noResultsMessage="No results found."
        onChange={[Function]}
        openOnFocus={true}
        options={
          Array [
            Object {
              "text": "New",
              "value": "created",
            },
            Object {
              "text": "Trending",
              "value": "trending",
            },
            Object {
              "text": "Hot",
              "value": "hot",
            },
            Object {
              "text": "Active",
              "value": "active",
            },
            Object {
              "text": "Promoted",
              "value": "promoted",
            },
            Object {
              "text": "Author",
              "value": "author",
            },
          ]
        }
        renderLabel={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        value="author"
        wrapSelection={true}
      />
    </div>
    <div
      className="authorName"
    >
      <Icon
        as="i"
        name="at"
      />
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        deburr={false}
        icon="dropdown"
        minCharacters={1}
        noResultsMessage="No results found."
        onChange={[Function]}
        onSearchChange={[Function]}
        openOnFocus={true}
        options={Array []}
        placeholder="username..."
        renderLabel={[Function]}
        search={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        selection={true}
        value=""
        wrapSelection={true}
      />
    </div>
    <div
      className="markyTag"
    >
      <Checkbox
        checked={false}
        disabled={true}
        label="Mark first tag as category"
        onChange={[Function]}
        type="checkbox"
      />
      <Popup
        content="Return search results which match the given first tag as their categories."
        keepInViewPort={true}
        on="hover"
        position="top left"
        trigger={
          <Icon
            as="i"
            name="question circle outline"
          />
        }
      />
    </div>
  </div>
  <div
    className="postList"
  >
    <Connect(PostList) />
  </div>
  <ScrollButton
    delayInMs={16.66}
    scrollStepInPx={50}
  />
</Layout>
`;

exports[`Testing on unconnected Search Testing search function should set error state and display error if fetchPost returned with error 1`] = `
<Layout
  item="search"
>
  <div
    className="searchBar"
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      allowAdditions={true}
      closeOnBlur={true}
      deburr={false}
      fluid={true}
      icon="search"
      minCharacters={1}
      multiple={true}
      noResultsMessage="No results found."
      onAddItem={[Function]}
      onChange={[Function]}
      openOnFocus={true}
      options={
        Array [
          Object {
            "text": "life",
            "value": "life",
          },
          Object {
            "text": "photography",
            "value": "photography",
          },
          Object {
            "text": "kr",
            "value": "kr",
          },
          Object {
            "text": "steemit",
            "value": "steemit",
          },
          Object {
            "text": "art",
            "value": "art",
          },
          Object {
            "text": "yatch",
            "value": "yatch",
          },
          Object {
            "text": "crypto",
            "value": "crypto",
          },
          Object {
            "text": "cake",
            "value": "cake",
          },
          Object {
            "text": "chalk",
            "value": "chalk",
          },
          Object {
            "text": "zebra",
            "value": "zebra",
          },
        ]
      }
      placeholder="Insert your tags here. Or leave empty to search for all posts..."
      renderLabel={[Function]}
      search={true}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      selection={true}
      value={
        Array [
          "kr",
          "art",
        ]
      }
      wrapSelection={true}
    />
    <Button
      as="button"
      loading={false}
      onClick={[Function]}
      positive={true}
      role="button"
    >
       
      Search
    </Button>
  </div>
  <div
    className="searchOptionsBar"
  >
    <div>
      Filter tags by
       
       
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        deburr={false}
        icon="dropdown"
        inline={true}
        minCharacters={1}
        noResultsMessage="No results found."
        onChange={[Function]}
        openOnFocus={true}
        options={
          Array [
            Object {
              "text": "New",
              "value": "created",
            },
            Object {
              "text": "Trending",
              "value": "trending",
            },
            Object {
              "text": "Hot",
              "value": "hot",
            },
            Object {
              "text": "Active",
              "value": "active",
            },
            Object {
              "text": "Promoted",
              "value": "promoted",
            },
            Object {
              "text": "Author",
              "value": "author",
            },
          ]
        }
        renderLabel={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        value="created"
        wrapSelection={true}
      />
    </div>
    <div />
    <div
      className="markyTag"
    >
      <Checkbox
        checked={false}
        disabled={false}
        label="Mark first tag as category"
        onChange={[Function]}
        type="checkbox"
      />
      <Popup
        content="Return search results which match the given first tag as their categories."
        keepInViewPort={true}
        on="hover"
        position="top left"
        trigger={
          <Icon
            as="i"
            name="question circle outline"
          />
        }
      />
    </div>
  </div>
  <div
    className="postList"
  >
    <Notifier
      message="Error: some kinda error"
      type="ERROR"
    />
  </div>
  <ScrollButton
    delayInMs={16.66}
    scrollStepInPx={50}
  />
</Layout>
`;

exports[`Testing on unconnected Search should render Search page correctly 1`] = `
<Layout
  item="search"
>
  <div
    className="searchBar"
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      allowAdditions={true}
      closeOnBlur={true}
      deburr={false}
      fluid={true}
      icon="search"
      minCharacters={1}
      multiple={true}
      noResultsMessage="No results found."
      onAddItem={[Function]}
      onChange={[Function]}
      openOnFocus={true}
      options={
        Array [
          Object {
            "text": "life",
            "value": "life",
          },
          Object {
            "text": "photography",
            "value": "photography",
          },
          Object {
            "text": "kr",
            "value": "kr",
          },
          Object {
            "text": "steemit",
            "value": "steemit",
          },
          Object {
            "text": "art",
            "value": "art",
          },
          Object {
            "text": "yatch",
            "value": "yatch",
          },
          Object {
            "text": "crypto",
            "value": "crypto",
          },
          Object {
            "text": "cake",
            "value": "cake",
          },
          Object {
            "text": "chalk",
            "value": "chalk",
          },
          Object {
            "text": "zebra",
            "value": "zebra",
          },
        ]
      }
      placeholder="Insert your tags here. Or leave empty to search for all posts..."
      renderLabel={[Function]}
      search={true}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      selection={true}
      value={Array []}
      wrapSelection={true}
    />
    <Button
      as="button"
      loading={false}
      onClick={[Function]}
      positive={true}
      role="button"
    >
       
      Search
    </Button>
  </div>
  <div
    className="searchOptionsBar"
  >
    <div>
      Filter tags by
       
       
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        deburr={false}
        icon="dropdown"
        inline={true}
        minCharacters={1}
        noResultsMessage="No results found."
        onChange={[Function]}
        openOnFocus={true}
        options={
          Array [
            Object {
              "text": "New",
              "value": "created",
            },
            Object {
              "text": "Trending",
              "value": "trending",
            },
            Object {
              "text": "Hot",
              "value": "hot",
            },
            Object {
              "text": "Active",
              "value": "active",
            },
            Object {
              "text": "Promoted",
              "value": "promoted",
            },
            Object {
              "text": "Author",
              "value": "author",
            },
          ]
        }
        renderLabel={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        value="created"
        wrapSelection={true}
      />
    </div>
    <div />
    <div
      className="markyTag"
    >
      <Checkbox
        checked={false}
        disabled={true}
        label="Mark first tag as category"
        onChange={[Function]}
        type="checkbox"
      />
      <Popup
        content="Return search results which match the given first tag as their categories."
        keepInViewPort={true}
        on="hover"
        position="top left"
        trigger={
          <Icon
            as="i"
            name="question circle outline"
          />
        }
      />
    </div>
  </div>
  <div
    className="postList"
  >
    <Connect(PostList) />
  </div>
  <ScrollButton
    delayInMs={16.66}
    scrollStepInPx={50}
  />
</Layout>
`;
